scenes/music_puzzle/music_puzzle.gd

```diff
@@ -0,0 +1,61 @@
+extends Node2D
+
+@export var xylophone: MusicalRockXylophone
+
+## Melodies expressed with the letters ABCDEFG.
+@export var melodies: Array[String]
+
+## A fire corresponding to each melody, ignited when the correct melody is played (in order).
+@export var fires: Array[Bonfire]
```

Comment:
This is totally not necessary!, but, a cool thing godot lets us do is add warnings to let devs know if a node isn't set up correctly. If we want this node to be modified by learners, we could also add an in-editor warning in case the melodies and fires don't add up:
```gdscript
func _get_configuration_warnings() -> PackedStringArray:
	if melodies.size() != fires.size():
		return ["There should be one fire for each melody, but now there are %d melodies and %d fires" % [melodies.size(), fires.size()]]
	return []
```

Note: the script needs to be a `@tool` for this to work!

![image](https://github.com/user-attachments/assets/75988c7b-e364-4707-9f9c-ac9a3178e24d)

Reply:
Good idea! I applied this suggestion with a tweak to the function to work around an edge case of `gdformat` that I don't like...